server/pulp/server/controllers/repository.py

```diff
@@ -82,6 +83,47 @@ def get_unit_model_querysets(repo_id, model_class, repo_content_unit_q=None):
         yield model_class.objects(id__in=chunk)
 
 
+def get_repo_unit_models(repo_id):
+    """
+    Retrieve all the MongoEngine models for units in a given repository. If a unit
+    type is in the repository and does not have a MongoEngine model, that unit type
+    is excluded from the returned list.
+
+    :param repo_id: ID of the repo whose unit models should be retrieved.
+    :type  repo_id: str
+
+    :return: A list of sub-classes of ContentUnit that define a unit model.
+    :rtype:  pulp.server.db.model.ContentUnit
+    """
+    unit_types = model.RepositoryContentUnit.objects(
+        repo_id=repo_id).distinct('unit_type_id')
+    unit_models = [plugin_api.get_unit_model_by_id(type_id) for type_id in unit_types]
+    # Filter any non-MongoEngine content types.
+    return filter(None, unit_models)
+
+
+def get_mongoengine_repo_querysets(repo_id, repo_content_unit_q=None):
```

Comment:
This is picky, but note that the name here is effectively ``repo_controller.get_mongoengine_repo_querysets``, and nowhere does the word "unit" appear. Of course units are the only thing we want out of these querysets.

This function is in the repo controller, so you don't necessarily need "repo" in the function name. Perhaps just replace it with  the word "unit", to make ``repo_controller.get_mongoengine_unit_querysets``